generator/gapic_generator_unittest.cc

```diff
@@ -85,16 +64,17 @@ TEST(CppGapicPluginTest, GapicPluginTest) {
       "google/example/library/v1/library_service_stub.gapic.cc",
   };
 
-  for (std::string const& f : files_to_check) {
-    std::string actual_file = absl::StrCat(gapic_out_dir, "/", f);
-    std::string expected_file =
-        absl::StrCat(test_data_dir, "/", f, ".baseline");
+  for (auto const& file : files_to_check) {
+    std::string const expected_file(data_dir + file + ".baseline");
+    std::string const actual_file(output_dir + file);
 
-    std::string actual_file_content = LoadContent(actual_file);
-    std::string expected_file_content = LoadContent(expected_file);
+    std::string const& expected_file_content = LoadContent(expected_file);
```

Comment:
Why change to a const&?

Reply:
The strings are not supposed to be changed. Since the strings contain actual file content (i.e. really long strings) it would make sense to try to help compiler to avoid unnecessary copying (I believe it depends on compiler what it does, and most likely gcc does everything efficiently for all possible assignements here, but since i was already going east const, I went there everywhere...

Reply:
Any modern compiler will elide the copy and either use move semantics for the return or do the return value optimization.